Source/WebCore/history/CachedPage.cpp

```diff
@@ -172,14 +173,16 @@ void CachedPage::restore(Page& page)
 #endif
 
     if (m_needsUpdateContentsSize) {
-        if (auto* frameView = localMainFrame->view())
+        if (auto* frameView = dynamicDowncast<LocalFrameView>(mainFrame.virtualView()))
```

Comment:
ditto here, though I think in the remote case it will case an empty function defined in ScrollableArea.h (likely the case above doesn't do much of anything either, but it feels wrong not calling in it since it's in the hierarchy.)

Reply:
`    virtual void updateContentsSize() { }
`